core/block/editor/smartblock/smartblock.go

```diff
@@ -639,8 +634,6 @@ func (sb *smartBlock) Apply(s *state.State, flags ...ApplyFlag) (err error) {
 		}
 	}
 
-	sb.beforeStateApply(s)
```

Comment:
This method calls sb.setRestrictionsDetail(s) and sb.injectLinksDetails(s). Both methods are also called in injectDerivedDetails in the beginning of function body